### ISSUE TITLE
Generic default deployment method for 0.6 release

### ISSUE THREAD

**Issue by krishnadurai** [kubeflow/kubeflow#3774]
**Issue Description:**
I am trying to follow the installation instructions in Kubeflow *kfctl* deployer with the command:
```
kfctl init {$KFAPP} --config=kfctl_k8s_istio.yaml
```

Config for deployment followed:
https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_k8s_istio.yaml#L51

Current base for mysql:
https://github.com/kubeflow/manifests/tree/86aff6edabeb6abc57bb93bad23a7e2952f8183c/pipeline/mysql/base

Current `mysqlPd` overlay:
https://github.com/kubeflow/manifests/tree/86aff6edabeb6abc57bb93bad23a7e2952f8183c/pipeline/mysql/overlays/mysqlPd

In the line for the config for deployment mentioned above, `mysqlPd` overlay is selected for being the PV for mySQL. This is the same behaviour for other components:
- katib-mysql
- metadata-mysql
- minio-pv-claim
- mysql-pv-claim

The `base` folder of the manifests doesn't have the config for PVs and PVCs required as seen in the link above. Which means, that it is upto the overlays to set it up.

**Logs:**
```
kubectl get pvc -n kubeflow
NAME             STATUS    VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS    AGE
katib-mysql      Pending                                      local-storage   8m37s
metadata-mysql   Pending                                      local-storage   8m37s
minio-pv-claim   Pending                                      local-storage   8m35s
mysql-pv-claim   Pending                                      local-storage   8m35s
```

**Proposed Solution**
First, we assume that a default storage class is already assigned in the Kubernetes cluster.
Using default storage class for PVCs required we should be able to overlay the PVCs. This will get it working with default storage class as mentioned in the Kubernetes best practises for writing portable configuration:
https://kubernetes.io/docs/concepts/storage/persistent-volumes/#writing-portable-configuration

@lluunn @jlewi @kunmingg @johnugeorge 

**Related Issues**
https://github.com/kubeflow/kubeflow/issues/3728
https://github.com/kubeflow/kubeflow/issues/3694

/assign @krishnadurai

**Comment by jlewi** [comment]
@krishnadurai Using a PVC makes sense to me.

But is the problem in the kustomize manifests or the overlays selected in 
https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_k8s_istio.yaml#L51

If I look at the base for mysql
https://github.com/kubeflow/manifests/blob/master/pipeline/mysql/base/persistent-volume-claim.yaml

There is a PVC defined and no PD. I think this is because @yanniszark already ran into this issue and fixed it in kubeflow/manifests#140; 

I believe in that PR he moved the PVC definition into the base. The PD overlays are then used to change that PVC to mount a specific PD which is a GCE PD.

So I think in kfctl_k8s_istio.yaml we just need to not apply those overlays.

Looking at
https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_k8s_istio.yaml#L51

Its not clear to me why it would be applying the [mySqlPd overlay](https://github.com/kubeflow/manifests/tree/master/pipeline/mysql/overlays/mysqlPd) since I don't see the overlay parameter specified.

It looks like 
kfctl_k8s_istio.yaml is still using the deprecated  KfDef.Spec.components, KfDef.Spec.component_params and KfDef.Spec.Packages. We should change that to just using KfDef.Spec.Applications; so that we can easily specify each kustomize package to install. I think that might fix the incorrect overlay being applied.

If you ran kfctl; it will update KfDef.Spec.Applications so you could just copy that into kfctl_k8s_istio.yaml to update it to use KfDef.Spec.Applications

@abhi-g Is this the same issue you ran into?

**Comment by abhi-g** [comment]
The issue looks similar to what I ran into. But I just tried a fresh deployment with the v1 config and seems to work without any issues. Last time when I ran into the issues with pd, I didn't specify the config to point to this yaml and rather left it unspecified.

**Comment by krishnadurai** [comment]
@abhi-g @jlewi  The v1 installation is working fine with fresh deployment without applying the overlay - I have tested this on an on-premise server. The confusion here was the [extra overlay parameters](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_k8s_istio.yaml#L48-L51) mentioned in the config file which led me to believe that this is applying the overlays for mysql and minio components.

The reason why these PVCs are in the pending state is that in an on-premise environment we need to have a dynamic volume provider like [local-path-provisioner](https://github.com/rancher/local-path-provisioner) which detects PVCs which are of default storage class type and provisions required PVs for them. This volume provisioner is available by default in GCP. As soon as we deployed this provisioner in our cluster, I was able to verify all components of Kubeflow are working as intended. We'll make a note of it in the deployment instructions for this config.

With this [fix PR](https://github.com/kubeflow/kubeflow/pull/3792), we have upgraded kfctl_k8s_istio.yaml's kfctl spec to v2 and removed unused parameters like the one mentioned above. The behaviour is the same as v1.
